src/In.ProjectEKA.HipService/DataFlow/CryptoHelper/CryptoHelper.cs

```diff
@@ -0,0 +1,116 @@
+            catch (Exception e)
+            {
+                Log.Fatal(e);
+                return "";
+            }
+        }
+        
+        private static IEnumerable<byte> GetDeriveKey(byte[] key1, AsymmetricCipherKeyPair senderKeyPair,
+            string curveName, string algorithm)
+        { 
+            var ecP = CustomNamedCurves.GetByName(curveName);
+            var ecSpec = new ECDomainParameters(ecP.Curve, ecP.G, ecP.N, ecP.H, ecP.GetSeed());
+            var publicKey = new ECPublicKeyParameters(ecSpec.Curve.DecodePoint(key1), ecSpec);
+            var agreement = AgreementUtilities.GetBasicAgreement(algorithm);
+            agreement.Init(senderKeyPair.Private);
+            var result = agreement.CalculateAgreement(publicKey);
+            return result.ToByteArrayUnsigned();
+        }
+
+        private static string Encrypt(string sharedKey, string dataToEncrypt,
+            string randomKeySender, string randomKeyReceiver)
+        {
+            var json = JsonConvert.SerializeObject(dataToEncrypt);
+            var xorOfRandoms = XorOfRandom(randomKeySender, randomKeyReceiver).ToArray();
+            var salt = xorOfRandoms.Take(20);
+            var iv = xorOfRandoms.TakeLast(12);
+            var hkdfBytesGenerator = new HkdfBytesGenerator(new Sha256Digest());
+            var hkdfParameters = new HkdfParameters(GetByteFromBase64(sharedKey).ToArray(),
+                salt.ToArray(),
+                null);
+            hkdfBytesGenerator.Init(hkdfParameters);
+            var aesKey = new byte[32];
+            hkdfBytesGenerator.GenerateBytes(aesKey, 0, 32);
+            try
+            {
+                return AesGcmEncryptor.EncryptDataUseAesGcm(json, aesKey, iv.ToArray());
+            }
+            catch (Exception e)
+            {
+                Log.Error("Error Occured while encryption {exception}: ",e);
+                return "";
+            }
+        }
+        
+        private static string GetBase64FromByte(IEnumerable<byte> value)
+        {
+            return Encoder.Base64.ToBase64String((byte [])value);
+        }
+
+        private static IEnumerable<byte> GetByteFromBase64(string value)
+        {
+            return Encoder.Base64.Decode(value);
+        }
+
+        public string GenerateRandomKey()
+        {
+            var rngCryptoServiceProvider = new RNGCryptoServiceProvider();
+            var randomBytes = new byte[32];
+            rngCryptoServiceProvider.GetBytes(randomBytes);
+            return GetBase64FromByte(randomBytes);   
+        }
+
+        private static IEnumerable<byte> XorOfRandom(string randomKeySender, string randomKeyReceiver)
+        {
+            var randomKeySenderBytes = GetByteFromBase64(randomKeySender).ToArray();
+            var randomKeyReceiverBytes = GetByteFromBase64(randomKeyReceiver).ToArray();
+            var sb = new byte[randomKeyReceiverBytes.Length];
+            for (var i = 0; i < randomKeySenderBytes.Length; i++)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'for' block.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5073772)